src/scheduler/buckets.cpp

```diff
@@ -1132,11 +1132,24 @@ check_node_buckets(status *policy, server_info *sinfo, queue_info *qinfo, resour
 			 * use that error code
 			 */
 			move_schd_error(err, failerr);
+	} else if (sinfo->svr_to_psets != NULL) {
+		int i;
+
+		/* Find buckets associated with nodes of the server which owns the job */
+		for (i = 0; sinfo->svr_to_psets[i] != NULL; i++) {
+			if (strcmp(sinfo->svr_to_psets[i]->svr_inst_id, resresv->job->svr_inst_id) == 0) {
+				nspec **nspecs;
+
+				nspecs = map_buckets(policy, sinfo->svr_to_psets[i]->np->bkts, resresv, err);
```

Comment:
this might have been deliberate to not confuse server_info's buckets with a node partition's .. I didn't write it originally, so I'd prefer leaving it as is unless you feel strongly about it.